File: pizzeria/pizzeria-core/src/main/java/io/github/joxit/pizzeria/model/Stock.java
package io.github.joxit.pizzeria.model;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import java.io.Serializable;

/**
 * @author Jones Magloire @Joxit
 * @since 2017-11-01
 */
@MappedSuperclass
public abstract class Stock<T extends Stockable> implements Serializable {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  protected int id;
  //    @ManyToOne
  protected T stockable;
  protected int quantity;
}

File: pizzeria/pizzeria-core/src/main/kotlin/io/github/joxit/pizzeria/model/Stock.kt
package io.github.joxit.pizzeria.model

import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.io.Serializable

@MappedSuperclass
abstract class Stock<T : Stockable>(
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  protected val id: Int,
  // @ManyToOne
  protected val stockable: T,
  protected val quantity: Int
) : Serializable